src/node/gov.h

```diff
@@ -5,26 +5,58 @@
 #include "ds/json.h"
 #include "entities.h"
 #include "kv/map.h"
+#include "proposals.h"
 
 namespace ccf
 {
   namespace jsgov
   {
     using ProposalId = std::string;
-    using Proposal = std::string;
+
     struct ProposalInfo
     {
       ccf::MemberId proposer_id;
+      ccf::ProposalState state;
       std::unordered_map<ccf::MemberId, std::string> ballots = {};
     };
-    DECLARE_JSON_TYPE(ProposalInfo)
-    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, ballots);
+    DECLARE_JSON_TYPE(ProposalInfo);
+    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, state, ballots);
+
+    struct ProposalSubmitted
+    {
+      ProposalId proposal_id;
+      ccf::ProposalState state;
+    };
+    DECLARE_JSON_TYPE(ProposalSubmitted);
+    DECLARE_JSON_REQUIRED_FIELDS(ProposalSubmitted, proposal_id, state);
 
-    using ProposalMap = kv::RawCopySerialisedMap<ProposalId, Proposal>;
+    using ProposalMap = kv::RawCopySerialisedMap<ProposalId, std::string>;
     using ProposalInfoMap = kv::MapSerialisedWith<
       ProposalId,
       ProposalInfo,
       kv::serialisers::BlitSerialiser,
       kv::serialisers::JsonSerialiser>;
+
+    struct Action
+    {
+      std::string name;
+      nlohmann::json args;
+    };
+    DECLARE_JSON_TYPE(Action);
+    DECLARE_JSON_REQUIRED_FIELDS(Action, name, args);
+
+    struct Proposal
+    {
+      std::vector<Action> actions;
+    };
+    DECLARE_JSON_TYPE(Proposal);
+    DECLARE_JSON_REQUIRED_FIELDS(Proposal, actions);
+
+    struct Ballot
```

Comment:
I've tried to create, and would like to enforce a distinction between `ballot`, the script submitted by a member, and `vote`, the outcome of that ballot applied to a `(proposal, kv)`. Thoughts?

Reply:
Sticking to that for now, we agree about the distinction, although `vote` may get renamed to something better.